### PR TITLE
connect after event handlers are added, fix tests

### PR DESCRIPTION
**Pull request by calvinfo**
This commit introduces several changes

- fixes tests by randomizing topic names
- bumps the circle nsq version
- adds reader connection post after construction

This last bullet allows the reader to add event handlers before
the connection is made.

### CODE REVIEW

**Comment by calvinfo** [comment]
The reason the tests are flakey with a single topic name is that I'm pretty sure the nsqlookups don't atomically delete the topic over the course of the request. It's quick enough on the local machine, but on circle it makes for a much flakier set of tests.

File: test/acceptance/reader.js [hunk 1]
```diff
@@ -2,203 +2,215 @@
...
-
-      pub.on('ready', function(){
-        pub.publish('test', 'something');
-      });
-
-      sub.once('message', function(msg){
-        msg.finish();
-        done();
-      });
+  afterEach(function(done){
```
Comment (juliangruber) [inline test/acceptance/reader.js]: do we need the `afterEach` when there's already the same logic in the `beforeEach`?

**Comment by juliangruber** [comment]
maybe i missed it in the diff, but can you add a test for "adds reader connection post after construction"?

File: test/acceptance/reader.js (hunk 1 above)
Comment (calvinfo) [inline test/acceptance/reader.js]: Yeah, good call. I moved them all to be `afterEach` since the topic name is created randomly now.

**Comment by calvinfo** [comment]
@juliangruber, good call–done!

**Comment by gjohnson** [comment]
LGTM!

**Comment by calvinfo** [comment]
thanks guys!

**Pull request merged**